### PR TITLE
To Do list: add & remove

### PR DESCRIPTION
**Pull request by fmanimashaun**
# Page add & remove functionality

This is the continuation of the #11  project. Here, I have added add and remove functionality to the app. Below are all the project requirements met:

- [x] Implement a function for adding a new task (add a new element to the array).
- [x] Implement a function for deleting a task (remove an element from the array).
- [x] Implement a function for editing task descriptions.
- [x] By default new tasks should have the property completed set to false and the property index set to the value of the new array length (i.e. if you're adding a 5th task to the list, the index of that task should equal 5).
- [x] Deleting a task updates all remaining items' indexes, so they represent the current list order and are unique(i.e. if we delete the first task index 1 from the list, the index of the next task(2) will be set to 1).
- [x] All changes to the To Do List should be saved in local storage.

[APP DEMO](https://fmanimashaun.github.io/To-do-List/)

[demo video](https://www.loom.com/share/9416bb2c3a654d5784fb61b0afce0533)

## Screenshot
image

### CODE REVIEW

File: src/js/modules/addTask.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  // add the new task to the task list
+  tasks.push(task);
+
+  // save the task list to local storage
+  localStorage.setItem('tasks', JSON.stringify(tasks));
+
+  // display the updated task list
+  displayTodos();
+};
+
```
Comment (wayungi) [inline src/js/modules/addTask.js:23]: - [ ] Good job adding the addTask functionality. Please make sure that the task indexes are number from 1 ... such that the last added task has the index equal to the length of the array as indicated in your PR body
pr

 Currently, your tasks are indexed from 0 as below.
missed

Comment (wayungi) [reply src/js/modules/addTask.js:23]: - [ ] It appears like you did not follow the instructions fully. `Deleting a task should update all remaining items' indexes, so they represent the current list order and are unique(i.e. if you're deleting the first task index 1 from the list, the index of the next task(2) should set to 1).` Below is the index numbering when I delete an Item
index
- [ ] Your id field is doing exactly what the index field should do. please refactor your code make the id field the index. 👍🏾

**Review by wayungi — CHANGES_REQUESTED** [review]
Hi @fmanimashaun , 
image

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- [ ] All linter checks passing ✔️ 
- [ ] Good commit messages ✔
- [ ] No merge conflicts ✔️ 
- [ ]  Correct GitHub flow ✔️ 
- [ ] Descriptive PR body 

## Required Changes ♻️
_Check the comments under the review._
- [ ] Conspicuous by its absence is the `LICENCE`. Please add one using this [resource](https://docs.github.com/en/communities/setting-up-your-project-for-healthy-contributions/adding-a-license-to-a-repository)
## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/js/modules/addTask.js (hunk 1 above)
Comment (fmanimashaun) [reply src/js/modules/addTask.js:23]: @wayungi 

this is the screenshot from my localstorage!
image

Comment (fmanimashaun) [reply src/js/modules/addTask.js:23]: @wayungi 

Kindly find the link to the demo video below:

[demo video](https://www.loom.com/share/9416bb2c3a654d5784fb61b0afce0533)

File: src/js/modules/editTask.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  task.description = str;
+
+  // save the updated task list to local storage
+  localStorage.setItem('tasks', JSON.stringify(tasks));
+
+  // display the updated task list
+  displayTodos();
+};
+
+export default editTask;
```
[the remark this answers is not in the document]
Comment (fmanimashaun) [reply src/js/modules/editTask.js:20]: @wayungi thanks, I have included a demo video link

File: LICENSE [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+The above copyright notice and this permission notice shall be included in all
+copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
```
Comment (tresorsawasawa) [inline LICENSE:21]: - [ **OPTIONAL** ] **RECOMMENDED:** Kindly make sure you update your README by using the correct template provided by using the correct README then you can customize all sections it according to your project.

**Review by tresorsawasawa — APPROVED** [review]
# STATUS: APPROVE :heavy_check_mark: :heavy_check_mark:  :champagne: 

Hi @fmanimashaun 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 :clap: :clap: :100: 

nodding-yes

## To highlight :champagne: 
- All linters are passing :heavy_check_mark: 
- Good commit history :heavy_check_mark: 
- Good PR description :heavy_check_mark: 

## Optional suggestions
_Every comment with the **[OPTIONAL]** prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

_Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please ping me [@tresorsawasawa](https://github.com/tresorsawasawa) when you comment so I can receive the notification or use slack with the same name._

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**